Detectors/TOF/calibration/src/TOFChannelCalibrator.cxx

```diff
@@ -338,8 +345,314 @@ float TOFChannelData::integral(int ch) const
 {
   // calculates the integral along one fixed channel and in the full x-range
 
-  return integral(ch, ch, 0, mNBins - 1);
+  return mEntries.at(ch);
+}
+
+//-------------------------------------------------------------------
+// TOF Channel Calibrator
+//-------------------------------------------------------------------
+
+template <typename T>
+void TOFChannelCalibrator<T>::finalizeSlotWithCosmics(Slot& slot)
+{
+  // Extract results for the single slot
+
+  o2::tof::TOFChannelData* c = slot.getContainer();
+  LOG(INFO) << "Finalize slot for calibration with cosmics " << slot.getTFStart() << " <= TF <= " << slot.getTFEnd();
+
+  // for the CCDB entry
+  std::map<std::string, std::string> md;
+  TimeSlewing& ts = mCalibTOFapi->getSlewParamObj(); // we take the current CCDB object, since we want to simply update the offset
+
+  int nbins = c->getNbins();
+  float range = c->getRange();
+  std::vector<int> entriesPerChannel = c->getEntriesPerChannel();
+
+#ifdef WITH_OPENMP
+  if (mNThreads < 1) {
+    mNThreads = std::min(omp_get_max_threads(), NMAXTHREADS);
```

Comment:
how many sectors does TOF have? Is it really 20?
Since you parallelize over the sectors, I think it would make sense to set NMAXTHREADS to the number of tof sectors?

Reply:
They are 18. The 20 is the number that I got on my desktop. I can change it to 18, sure.

Reply:
If you want to impose a limit, I would put nSectors, 20 doesn't make sense since with parallelization over the sectors you cannot run 20 threads.
But in principle you also do not need to impose a limit

Reply:
Yes, you are right, only 18 threads will be created at max in any case. 20 was just the limit I had on my desktop, so what `omp_get_max_threads()` was returning. Then I thought that if you have something else running, this number will be less, so I used the minimum between the two. But if this is exactly what is done automatically, then I would remove this setting.

Reply:
So, now I checked the code. Here the logic is: if you don't set the number of threads from outside, then you want that it parallelize as much as it can. If I just put:
` mNThreads = omp_get_max_threads();`
I guess it will know that the max is 18 from the fact that we parallelize the loop over the 18 sectors, right?